Sources/NIOCore/ByteBuffer-hexdump.swift

```diff
@@ -0,0 +1,188 @@
+    /// so you should set those indices to desired location to get the offset and length that you need to dump.
+    func hexDumpShort() -> String {
+        var hexString = ""
+        hexString.reserveCapacity(self.readableBytes * 3)
+
+        for byte in self.readableBytesView {
+            hexString += String(byte: byte, padding: 2)
+            hexString += " "
+        }
+
+        return String(hexString.dropLast())
+    }
+
+    /// Return a `String` of hexadecimal digits of bytes in the Buffer.,
+    /// in a format that's compatible with `xxd -r -p`, but clips the output to the max length of `limit` bytes.
+    /// If the dump contains more than the `limit` bytes, this function will return the first `limit/2`
+    /// and the last `limit/2` of that, replacing the rest with `...`, i.e. `01 02 03 ... 09 11 12`.
+    ///
+    /// - parameters:
+    ///     - limit: The maximum amount of bytes presented in the dump.
+    func hexDumpShort(limit: Int) -> String {
+        guard self.readableBytes > limit else {
+            return self.hexDumpShort()
+        }
+
+        var buffer = self
+        let front = buffer.readSlice(length: limit / 2)!
+        buffer.moveReaderIndex(forwardBy: buffer.readableBytes - limit / 2)
+        let back = buffer.readSlice(length: buffer.readableBytes)!
+
+        let startHex = front.hexDumpShort()
+        let endHex = back.hexDumpShort()
+        return startHex + " ... " + endHex
+    }
+
+    // Dump hexdump -C compatible format without the last line containing the byte length of the buffer
+    func _hexDumpLines(offset: Int) -> String {
+        var result = ""
+        // Each hex dump line represents 16 bytes, and takes up to 79 characters including the \n in the end.
+        // Plus the last line of the hex dump which is just 8 characters.
+        result.reserveCapacity(self.readableBytes / 16 * 79 + 8)
+
+        var buffer = self
+        // We're going to move the lineOffset index forward, so copy it into a variable
+        var lineOffset = offset
+
+        while buffer.readableBytes > 0 {
+            let slice = buffer.readSlice(length: min(16, buffer.readableBytes))!
+            let lineLength = slice.readableBytes
+
+            // Left column of the hex dump signifies the offset from the beginning of the dumped buffer
+            // and is separated from the next column with two spaces.
+            result += String(byte: lineOffset, padding: 8)
+            result += "  "
+
+            // Center column consists of:
+            // - xxd-compatible dump of the first 8 bytes
+            // - space
+            // - xxd-compatible dump of the rest 8 bytes
+            // If there are not enough bytes to dump, the column is padded with space.
+            let left = slice.getSlice(at: 0, length: min( 8, slice.readableBytes ))!
+            let right = slice.getSlice(at: 8, length: min( 8, slice.readableBytes - 8)) ?? ByteBuffer()
```

Comment:
Minor thing here, we can avoid `getSlice` again by grabbing a mutable copy of the slice:

```swift
var otherSlice = slice
let left = otherSlice.readSlice(length: min(8, otherSlice.readableBytes))!
let right = otherSlice.readSlice(length: min(8, otherSlice.readableBytes))!
```